Source/JavaScriptCore/API/glib/JSCValue.cpp

```diff
@@ -2145,3 +2145,30 @@ char* jsc_value_to_json(JSCValue* value, unsigned indent)
 
     return json;
 }
+
+/**
+ * jsc_value_new_deferred_promise:
+ * @context: a #JSCContext
+ * @value: a #JSCValue
+ *
+ * Create a new Promise returning handles to resolve and reject method
+ *
+ * Returns: (transfer full): a defered promise
+ *
+ * Since: 2.48
+ */
+JSCValue* jsc_value_new_deferred_promise(JSCContext* context, JSCValue **resolve, JSCValue **reject)
+{
+    g_return_val_if_fail(JSC_IS_CONTEXT(context), nullptr);
+    auto* jsContext = jscContextGetJSContext(context);
+    JSObjectRef resolveObj, rejectObj;
+    JSCValue* ret;
+
+    ret = jscContextGetOrCreateValue(context, JSObjectMakeDeferredPromise(jsContext, &resolveObj, &rejectObj, nullptr)).leakRef();
```

Comment:
We want to pass `&resolveObj` and `&rejectObj` only if `resolve` and `reject` are non-null, respectively, to avoid having to destroy them immediately after creation in case the caller will not be using them. Also, the call may be all in one line:

```suggestion
    JSObjectRef resolveObj = nullptr;
    JSObjectRef rejectObj = nullptr;
    JSCValue* ret = jscContextGetOrCreateValue(context, JSObjectMakeDeferredPromise(jsContext, resolve ? &resolveObj : nullptr, reject ? &rejectObj : nullptr, nullptr)).leakRef();
```

Reply:
@carlosgcampos This is ignoring the `exception` argument to `JSObjectMakeDeferredPromise()`, do you know if that would be okay here?

Reply:
Yes, exception should be handled